docs/elasticsearch.rst

```diff
@@ -83,7 +83,74 @@ blow away the index and start over.
 .. _elasticsearch-head: https://github.com/mobz/elasticsearch-head
 
 
-Querying Elasticsearch
-----------------------
+Querying Elasticsearch - Best Practices
+---------------------------------------
+
+Here are the most basic things to know if you want to write readable
+and reasonably performant code for accessing Elasticsearch.
+
+
+Use ESQuery when possible
+=========================
 
 Check out :doc:`/es_query`
+
+ * Prefer the cleaner ``.count()``, ``.values()``,  ``.values_list()``, etc. execution methods
+   to the more low level ``.run().hits``, ``.run().total``, etc.
+   With the latter easier to make mistakes and fall into anti-patterns and it's harder to read.
+ * Prefer adding filter methods to using ``set_query()``
+   unless you really know what you're doing and are willing to make your code more error prone
+   and difficult to read.
+
+
+Prefer "get" to "search"
+========================
+
+Don't use search to fetch a doc or doc fields by doc id; use "get" instead.
+Searching by id can be easily an order of magnitude (10x) slower. If done in a loop,
+this can effectively grind the ES cluster to a halt.
+
+**Bad:**::
+
+    POST /hqcases_2016-03-04/case/_search
+    {
+      "query": {
+        "filtered": {
+          "filter": {
+            "and": [{"terms": {"_id": [case_id]}}, {"match_all": {}}]
+          },
+          "query": {"match_all":{}}
+        }
+      },
+      "_source": ["name"],
+      "size":1000000
+    }
+
+**Good:**::
+
+    GET /hqcases_2016-03-04/case/<case_id>?_source_include=name
```

Comment:
Might be worth giving some code samples instead of this format. ESQuery is my go-to way of interacting with ElasticSearch (and I think we encourage that, and discourage building the requests manually), so it might be worth pointing out that this is an exception to that rule, and showing which function should be used instead (is it `get_es_new().get(index, id)`?)